.github/workflows/ci.yml

```diff
@@ -0,0 +1,49 @@
+# Copyright (c) 2023 Robert Bosch GmbH and Microsoft Corporation
+#
+# This program and the accompanying materials are made available under the
+# terms of the Apache License, Version 2.0 which is available at
+# https://www.apache.org/licenses/LICENSE-2.0.
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+# WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+# License for the specific language governing permissions and limitations
+# under the License.
+#
+# SPDX-License-Identifier: Apache-2.0
+
+name: CI
+concurrency:
+  group: ${{ github.ref }}
+  cancel-in-progress: true
+
+on:
+  workflow_dispatch:
+  push:
+    # Run only on branches/commits and not tags
+    branches:
+      - main
+  pull_request:
+    branches:
+      - main
+
+jobs:
+  coverage:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - uses: actions/setup-node@v3
+        with:
+          node-version: 20.4.0
```

Comment:
please use version 18 to be in synced with the other workflow

Reply:
From security point of view - the most latest version in use - less potential security issues appeared.
Therefore it is ok to use it, but consider the latest possible... From other side: officially still 18,17.0 is considered as latest stable: https://nodejs.org/en/download

Reply:
it is not really about security but about consistency. We are using 18 in the repo and the other workflow and to be aligned and not have to maintain another setup we should set everything to 18